cfme/common/vm.py

```diff
@@ -448,6 +455,125 @@ def instantiate(self, name, provider, template_name=None):
+                if provision_request.is_succeeded(method='ui'):
+                    logger.info('Waiting for vm %s to appear on provider %s', vm.name,
+                                provider.key)
+                    wait_for(provider.mgmt.does_vm_exist, [vm.name],
+                             handle_exception=True, num_sec=600)
+                else:
+                    logger.warn("Provisioning failed with the message {}".format(
+                        provision_request.row.last_message.text))
+                    return None
+        if find_in_cfme:
+            vm.wait_to_appear(timeout=800)
+
+        return vm
+
+    def create_rest(self, vm_name, provider, form_values=None, check_existing=False):
+        """
+        Provisions a VM/Instance with the default self.vm_default_args_rest.
+        self.vm_default_args_rest may be overridden by form_values.
+        For more details about rest attributes please check:
+        https://access.redhat.com/documentation/en-us/red_hat_cloudforms/4.6/html-single/
+        red_hat_cloudforms_rest_api/index#provision-request-supported-attributes or
+        http://manageiq.org/docs/reference/fine/api/appendices/provision_attributes
+        NOTE: placement_auto defaults to True for requests made from the API or CloudForms Automate.
+        Args:
+            vm_name: vm name
+            provider: provider object
+            form_values: overrides default provision arguments or extends it.
+            check_existing: cancel creation if VM exists
+        Return: Instance object
+        """
+        vm = self.instantiate(vm_name, provider)
+        if check_existing and vm.exists:
+            vm
+        else:
+            if not provider.is_refreshed():
+                provider.refresh_provider_relationships()
+                wait_for(provider.is_refreshed, func_kwargs={'refresh_delta': 10}, timeout=600)
+
+            if not form_values:
+                form_values = vm.vm_default_args_rest
+            else:
+                inst_args = vm.vm_default_args_rest
+                form_values = recursive_update(inst_args, form_values)
+            response = self.appliance.rest_api.collections.provision_requests.action.create(
+                **form_values)[0]
+            assert_response(vm.appliance)
+
+            provision_request = vm.appliance.collections.requests.instantiate(
+                description=response.description)
+
+            provision_request.wait_for_request(num_sec=900)
+            if provision_request.is_succeeded():
+                wait_for(lambda: provider.mgmt.does_vm_exist(vm.name), num_sec=1000, delay=5,
+                         message="VM {} becomes visible".format(vm.name))
+            else:
+                logger.error("Provisioning failed with the message {}".
+                            format(provision_request.rest.message))
+                raise Exception(provision_request.rest.message)
```

Comment:
**REQUIRED**
We have https://github.com/ManageIQ/integration_tests/blob/master/cfme/exceptions.py#L5